bpf.go

```diff
@@ -317,6 +323,67 @@ func constructFilter(subsys Table, filters []ConnAttr) ([]bpf.RawInstruction, er
 	return raw, nil
 }
 
+func filterMarkAttribute(filters []ConnAttr) []bpf.RawInstruction {
```

Comment:
Thanks for your contribution! :pray: 

As `filterMarkAttribute()` is similar to `filterAttribute()` I'm wondering if both functions should be combined into one. What do you think?

> I'm testing it and it seems to be working fine. Please let me know what do you think about it. If you say it's ok, I will add tests which will cover this filter.

Adding a test would be great :+1:

Reply:
@dvomartin are you still working on this?

Reply:
Hi @florianl,
sorry, I was sick and forgot about this :) 

I can try to prepare second merger request which will adjust original function(s), but I'm afraid there will be too many ifs.

Reply:
Thanks @dvomartin for https://github.com/florianl/go-conntrack/pull/56 and hope you're fine again.

would you mind sharing your testcase? I'm fine with refactoring code later on with this (https://github.com/florianl/go-conntrack/pull/54) PR applied, if we can make sure with a testcase, that things don't break.

Reply:
Sure,
I have added some tests. My tests are slightly different - instructions are visible and invalid instructions in failed tests are printed to quickly find where is the problem.
I also added possibility to enable debug option to Nfct object which will print bpf instructions if enabled.

Print format is the same as in original C library to quickly compare results.

PR is still marked as WIP. It should be OK as it is, but I want to verify some things before I create final PR.

Reply:
thanks for the update!
please feel free to mark the PR as ready for review, if you are.

Reply:
It should be ready. Please check the changes.
I can update your original tests to the same format in separate PR, if you want.